### PR TITLE
accuknox data agent installation guides

### PR DESCRIPTION
**Pull request by sudhakaran19**

### CODE REVIEW

File: docs_sample/data-protection/data-agent/introduction.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## AccuKnox Data Agent(Object scanner)
+Accuknox Data Agent should have an object scanner.
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/introduction.md:2]: Describe what AccuKnox Data Agent is. We do not need to define what object scanner is.

**Comment by saleem-accuknox** [comment]
Everywhere, only refer as AccuKnox Data Agent (ADA). Do not refer as object scanner.
Let's assume that we are only writing the docs for MySQL
Describe what ADA is.
Describe how ADA works with MySQL.
Describe permissions required by ADA on MySQL DBs.
Describe the steps to download (Assume that we ship the binary in tar.gz format)
Describe the installation and configuration steps
Do not use go run command to run. 

We need to build it, ship the binary output to user in tar.gz format.
We will need to execute the ada binary in the background. So, write appropriate command

File: docs_sample/data-protection/data-agent/requirements.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Accuknox Data Agent scans for the database, tables in the database and columns in those tables.
+The Accuknox Data Agent uploads the data to Accuknox Control Plane. Accuknox Control Plane pushes the message to a Kafka topic
+
+## How Accuknox Data Agent work with MySQL
+Accuknox Data Agent get MySQL connection(user, host, port, password) related details from conf/agent-db-config.yaml. Then ADA get Mysql instance and get All DB, Tables and Columns details.
+
+## Permissions required by ADA on MySQL DBs.
+Which user created and configured in conf/agent-db-config. Give grant all permission to that user. Then only Accuknox Data Agent get DB instance and DB details. 
+
+CREATE USER 'ada_test_user'@'localhost' IDENTIFIED BY 'password';
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/requirements.md:12]: Create a proper use. Not a test user.

Comment (saleem-accuknox) [reply docs_sample/data-protection/data-agent/requirements.md:12]: Add a suggestion telling user to provide secure password.

File: docs_sample/data-protection/data-agent/requirements.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+GRANT ALL ON *.* TO 'ada_test_user'@'localhost';
+
+## Accuknox Data Agent DB configuration
+AccuKnox Data Agent can scan more than 1 DataBase at a time.
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/requirements.md:16]:
```suggestion
AccuKnox Data Agent can scan more than 1 Database at a time. 
```

File: docs_sample/data-protection/data-agent/requirements.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+The database to be scanned is configured through a file located at conf/agent-db-config.yaml The following is an example of a agent-db-config.yaml:
+
+apiVersion: v1
+type: agent-db-config
+data: 
+  workspace: 148
+  apiToken: 
+  databases:
+    - version: V1
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/requirements.md:26]: What does version mean here, explain in comment

Comment (sudhakaran19) [reply docs_sample/data-protection/data-agent/requirements.md:26]: mysql version

File: docs_sample/data-protection/data-agent/requirements.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      port: 5432
+      user: test
+      password: test
+
+## Installation
+Unzip AccuKnox Data Agent
+
+Edit the conf/agent-db-config.yaml
+
+Configure the agent db config to be monitored
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/requirements.md:46]: This seems to be the same as above. Redundant

File: docs_sample/data-protection/data-agent/requirements.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+Add the workspace
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/requirements.md:48]: Workspace ID

File: docs_sample/data-protection/data-agent/requirements.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+Add the apiToken
+
+The app.yaml and agent-db-config.yaml files must be present inside conf/ folder.
+
+Run AccuKnox Data Agent
```
Comment (saleem-accuknox) [inline docs_sample/data-protection/data-agent/requirements.md:54]: Provide command to run.

**Comment by saleem-accuknox** [comment]
In requirement.md, add requirement of a Linux Host with Ubuntu, CentOS, Fedora, RHEL as the OS. Add minimum RAM and Storage requirements.
Since we will also be communicating with our backend, let the user know about firewall requirements.

Also, move the installation section to installation.md

File: docs_sample/data-protection/data-agent/requirements.md (hunk 5 above)
Comment (sudhakaran19) [reply docs_sample/data-protection/data-agent/requirements.md:46]: Will remove one.

File: docs_sample/data-protection/data-agent/requirements.md (hunk 2 above)
Comment (sudhakaran19) [reply docs_sample/data-protection/data-agent/requirements.md:12]: What do you mean by proper user?.

Comment (saleem-accuknox) [reply docs_sample/data-protection/data-agent/requirements.md:12]: Proper user name. Don’t use test in the name.

**Review by saleem-accuknox — APPROVED** [review]

**Pull request merged**